Split "setting up development" into a separate guide

MatMoore opened the pull request:
Move this from `CONTRIBUTING.md` to `docs/setting-up-development.md`, so that it's 
easier to understand and link to.

I've also reordered some of the sections in it to make it easier to get to the point where you have the app running.

See issue https://github.com/exercism/exercism.io/issues/3251

ErikSchierboom commented on docs/setting-up-local-development.md:
```diff
@@ -1 +1,286 @@
+
+- Node.js
+
+To install Ruby, check out [RVM](https://rvm.io), [rbenv](https://github.com/sstephenson/rbenv) or [ruby-install](https://github.com/postmodern/ruby-install).
```
Perhaps you could link directly to the installation documentation pages? For example, for RVM you could refer to https://rvm.io/rvm/install.

ErikSchierboom commented on docs/setting-up-local-development.md:
```diff
@@ -1 +1,286 @@
+
+To install Ruby, check out [RVM](https://rvm.io), [rbenv](https://github.com/sstephenson/rbenv) or [ruby-install](https://github.com/postmodern/ruby-install).
+
+PostgreSQL can be installed with [Homebrew](http://brew.sh) on Mac OS X: `brew install postgresql`
```
Two small things:

1. Perhaps you could make it a bulleted list, with an entry for each platform. 
2. I'd like to see a link to a Windows installation too.

ErikSchierboom commented on docs/setting-up-local-development.md:
```diff
@@ -1 +1,286 @@
+* Fetch the seed data: `rake db:seeds:fetch`
+* Seed the database: `rake db:seed`
+
+If you are having 'PG::Connection ...' or 'Peer authentication failed for user ...' issues you can follow these steps (this is assuming that you modified the database.yml file and your changes are not being accessed properly. Anywhere that it is stated "PGUSER=postgres" replace "postgres" with your specified username that you use for your postgresql databases):
```
I would put the error messages in backticks, to make them standout.

kytrinyx commented:
I'm stoked to see this change—the Contributing Guide has been an overwhelming wall of text and this definitely helps.

MatMoore commented:
Thanks for the review @ErikSchierboom. I've added a few more commits implementing your suggestions.

I've linked to [https://www.postgresql.org/download/windows/](https://www.postgresql.org/download/windows/) which has installers for windows (I haven't tried these myself but it seems like the obvious way to do it). For everything else we're already linking out to instructions that cover multiple platforms so I don't think I can improve on that.

I also split up the troubleshooting section a bit more. I think it's still a bit confusing which `db:` tasks do what and why, but hopefully that helps.

ErikSchierboom commented:
Great, LGTM!

kytrinyx commented:
Thank you, @MatMoore!
